src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -145,6 +145,16 @@ void GcsJobManager::AddJobFinishedListener(JobFinishListenerCallback listener) {
 void GcsJobManager::HandleGetAllJobInfo(rpc::GetAllJobInfoRequest request,
                                         rpc::GetAllJobInfoReply *reply,
                                         rpc::SendReplyCallback send_reply_callback) {
+  // Get all job info. This is a complex operation:
+  // 1. One GetAll from the job table.
```

Comment:
I think https://github.com/ray-project/ray/pull/46861 will help here as well

Reply:
surprised I haven't merged that; will debug now